kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProducerMetrics.java

```diff
@@ -116,34 +121,43 @@ final class ProducerMetrics {
   }
 
   ProducerMetrics(KafkaRestConfig config, Map<String, String> metricsTags) {
+    // Tag should only be empty or "tenant" but sort on the key just in case
+    SortedMap<String, String> sortedMetricsTags = new TreeMap<>(metricsTags);
+    String sensorTags =
+        sortedMetricsTags.keySet().stream()
+            .map(key -> ":" + metricsTags.get(key))
+            .collect(Collectors.joining());
```

Comment:
There are two types of things with names:

The Sensors, which are a grouping of metrics, and the metrics themselves.

metrics:
When you make a metric (eg an Avg, or a WindowedCount, or a ...) it is passed a HashMap of tag->values (ie the metrics tags).  These are what show up in the tag list for the metrics seen in the tests, and in jconsole.

eg I produce messages which generate metrics with two tags: otherTag=abc and tenant=1

<img width="581" alt="image" src="https://user-images.githubusercontent.com/87994408/183916473-4edbeab3-7116-4c5f-a5e7-56fa6bdf8966.png">

In the test case, that's the equivalent of the ProducerMetrics we make with  otherTag=otherValue2 and tag=value2.

These tags get put onto the metrics via the metricTags HashMap, which sets both the key and the value.  They show up as comma separated tags in the test case

sensors:
The Sensors don't really show up anywhere, they just need a unique name, so that unique lkcs are tracked separately at a metrics level.  If you log a request against a sensor you just want that request to show up for that lkc, not all lkcs.  So you need a sensor per lkc.  

The name looks like
<img width="481" alt="image" src="https://user-images.githubusercontent.com/87994408/183921183-9377f506-8ca9-4dba-a4c0-ee9ae27161f2.png">

I used : as that is what is already used in the sensor name as a delimiter.

I didn't think it was worth adding the key (tenant) to the unique name for the string.  The LKCs are all unique.  The flip side is that this is generic code, so other people could use the tags for something else, and so we should add in the key too.

I'd prefer to leave it as is, because all we need is uniqueness of name per LKC but if you really want tenant=123 in the name I can change it.

Reply:
And argg grrr gnash to 

```
   String sensorTags =
        sortedMetricsTags.keySet().stream()
```

It gets reverted to all be on one line when i run mvn package.

Reply:
The relation between the sensor cardinality, the resulting metrics cardinality, and the tags cardinality is something I don't understand in as much depth as I would have wanted to. E.g. I wonder if we do need splitting sensors by tags other than the cluster ID.

> I'd prefer to leave it as is, because all we need is uniqueness of name per LKC but if you really want tenant=123 in the name I can change it.

I think it would be nicer to use a consistent format across the names, and I think it should be a simple change. Having said that, feel free to continue as-is.